gef.py

```diff
@@ -6168,10 +6176,21 @@ def __init__(self) -> None:
     @only_if_gdb_running
     def do_invoke(self, _: List[str], **kwargs: Any) -> None:
         args = kwargs["arguments"]
-        for arena in gef.heap.arenas:
-            self.dump_chunks_arena(arena, print_arena=args.all, allow_unaligned=args.allow_unaligned)
-            if not args.all:
-                break
+        if args.all or not args.arena_address:
+            for arena in gef.heap.arenas:
+                self.dump_chunks_arena(arena, print_arena=args.all, allow_unaligned=args.allow_unaligned)
+                if not args.all:
```

Comment:
how would `args.all` change while in this loop? We are in an `if args.all` case.

Reply:
Actually, now that I fixed it I noticed (and remembered) that it did make sense before:
The loop condition is not `and` but `or`:
- if `args.all` is set:  the chunks for all arenas are displayed
- if `args.all` is not set and `args.arena_address` is not provided: the chunks for the selected arena are printed (but not for all arenas)

Reply:
you're right I read the logic wrong.